api/client/registry.go

```diff
@@ -91,6 +91,10 @@ func (cli *DockerCli) ConfigureAuth(flUser, flPassword, serverAddress string, is
 		cli.in = os.Stdin
 	}
 
+	if !isDefaultRegistry {
```

Comment:
> Would it make sense to remove the isDefaultRegistry flag (which doesn't seem to be set correctly), and instead have ConfigureAuth check if serverAddress == ElectAuthServer(ctx) instead?

yup, works for me, so that `IsDefaultRegistry` becomes a local variable and is set to `:= serverAddress == ElectAuthServer(ctx)` - am I understanding correctly?